installer/install.sh

```
@@ -0,0 +1,635 @@
+    CF_TUNNEL_TOKEN=$(curl -s -X GET "https://api.cloudflare.com/client/v4/accounts/$ACCOUNT_ID/cfd_tunnel/$TUNNEL_ID/token" \
+         -H "X-Auth-Email: $CF_EMAIL" \
+         -H "X-Auth-Key: $CF_API_KEY" \
+         -H "Content-Type: application/json" | jq -r '.result')
+
+    if [ "$CF_TUNNEL_TOKEN" == "null" ] || [ -z "$CF_TUNNEL_TOKEN" ]; then
+        echo "Error: Could not get Cloudflare Tunnel token. This might be due to API limitations."
+        echo "Please try the semi-automatic mode."
+        return 1
+    fi
+
+    ZONE_NAME=$(echo $CF_HOSTNAME | awk -F. '{print $(NF-1)"."$NF}')
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)